### ISSUE TITLE
Target_location_loss is always the same

### ISSUE THREAD

**Issue by agi1512** [liuruoze/mini-AlphaStar#11]
Hello. I wonder if you could help me. I parsed 100 ladder replays of matches for both sides (protoss v protoss only), in total 200 parsed files, version is 4.10.0. When training for over 6 epochs (600 iterations) all losses change (go down usually), but target_location_loss always stays at 11.090355 no matter what i do. And then if i try to launch RL training using replays (on total loss ~24) model just moves camera, no other commands issued. I'm just curious is it a bug, regarding versions of replays, or i'm doing something else wrong

**Comment by agi1512** [comment]
Also, one more thing i noticed, when i were processing replays, pickling is too space-demanding. I used below code to compress pickle files on the fly
```python
import lzma


class SaveType(enum.IntEnum):
    torch_tensor = 0
    python_pickle = 1
    numpy_array = 2
    pickle_lzma = 3
    
SAVE_TYPE = SaveType.pickle_lzma

...

# at the start of main loop of processing replays
# helps save time to not process same replay twice when restarted
savefile_name = os.path.join(FLAGS.save_path, replay_file.replace('.SC2Replay', '')) + f'_{FLAGS.observed_player}'
if SAVE_TYPE == SaveType.torch_tensor:
    savefile_name += '.pt'
elif SAVE_TYPE == SaveType.python_pickle:
    savefile_name += '.pickle'
elif SAVE_TYPE == SaveType.pickle_lzma:
    savefile_name += '.xz'
if os.path.exists(savefile_name):
    print('replay exists')
    j += 1
    continue

...

# at the "SAVE_TYPE" if statement

elif SAVE_TYPE == SaveType.pickle_lzma:
    with lzma.open(savefile_name, 'wb') as handle:
        pickle.dump(step_dict, handle, protocol=pickle.HIGHEST_PROTOCOL)
# this manages to save up to 99.5% space. For example i had replay that weighs over 680MB, it was compressed to 2700KB!

```

To read compressed picle you just need to replace `open` with `lzma.open` and it's done.

**Comment by agi1512** [comment]
If this information would be any help to you:
I tried recently to launch this agent (attempt on training on ladder matches) against itself on Simple64 map and it started issuing commands to probes and sometimes even attack opponents probes. But on regular maps from training set it just sits and moves camera indefinitely. Such a weird interaction.

**Comment by liuruoze** [comment]
Sorry for replying late. This error is considerably strange. 

In my test, the target_location_loss actually can decline when at training.

I will check that and will respond when I have some discovery.

**Comment by liuruoze** [comment]
Hi, @agi1512  the new version of mAS (1.04 now) may solve your problem (the loss calculation). In this version, we add the mask operator for supervised learning (making the not applied arguments don't add the calculation of loss), thus the learning will become more accurate. Moreover, we give a statistical of move camera accuracy and non-camera action accuracy, to show the learning conditions of the two action types.

We close this issue now. If you find your problems are not solved, or new problems arise, re-open this issue as you wish.
